client/data.go

```diff
@@ -76,21 +84,59 @@ func (c *GrpcClient) Search(ctx context.Context, collName string, partitions []s
 			offset += rc
 			continue
 		}
-		entry.Fields = make([]entity.Column, 0, len(fieldDataList))
-		for _, fieldData := range fieldDataList {
-			column, err := entity.FieldDataColumn(fieldData, offset, offset+rc)
-			if err != nil {
-				entry.Err = err
-				continue
-			}
-			entry.Fields = append(entry.Fields, column)
-		}
+		entry.Fields, entry.Err = c.parseSearchResult(schema, outputFields, fieldDataList, i, offset, offset+rc)
 		sr = append(sr, entry)
 		offset += rc
 	}
 	return sr, nil
 }
 
+func (c *GrpcClient) parseSearchResult(sch *entity.Schema, outputFields []string, fieldDataList []*schema.FieldData, idx, from, to int) ([]entity.Column, error) {
+	outputs := make(map[string]struct{})
+	for _, output := range outputFields {
+		outputs[output] = struct{}{}
+	}
+	dynamicField := sch.GetDynamicField()
+	fields := make(map[string]*schema.FieldData)
+	var dynamicColumn *entity.ColumnJSONBytes
+	for _, fieldData := range fieldDataList {
+		fields[fieldData.GetFieldName()] = fieldData
+		if dynamicField != nil && fieldData.GetFieldName() == dynamicField.Name {
+			column, err := entity.FieldDataColumn(fieldData, from, to)
+			if err != nil {
+				return nil, err
+			}
+			var ok bool
+			dynamicColumn, ok = column.(*entity.ColumnJSONBytes)
+			if !ok {
+				return nil, errors.New("dynamic field not json")
+			}
+		}
+	}
+	columns := make([]entity.Column, 0, len(outputFields))
+	for outputField := range outputs {
```

Comment:
This breaks the order of columns, which may should be aligned to `outputFields`, is it by design?

Reply:
Simply iterate on `outputFields` could solve it

Reply:
fixed